xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateStore.java

```diff
@@ -2845,20 +2841,7 @@ private SessionFactory injectCustomMappingsInSessionFactory(XWikiDocument doc, X
             return getSessionFactory();
         }
 
-        Configuration config = getConfiguration();
-        SessionFactoryImpl sfactory = (SessionFactoryImpl) config.buildSessionFactory();
-        Settings settings = sfactory.getSettings();
-        ConnectionProvider provider = ((SessionFactoryImpl) getSessionFactory()).getSettings().getConnectionProvider();
```

Comment:
The `connectionProvider` is not part of the settings anymore so setting it doesn't seem to be needed.